src/main/java/com/xpanxion/java/springboot/da1/demo/service/student1/GymService1.java

```diff
@@ -25,4 +40,53 @@ public void addGym(Gym1 gym) {
 
     }
 
+    public AtomicReference<String> addGymMember(int gymId, GymMember1 gymMember1) {
+
+        AtomicReference<String> result = new AtomicReference<>("Gym Added!");
+
+        gymMember1.setGymId(gymId);
+        Optional<Gym1> gym = gymRepository1.findById(gymId);
+        gym.ifPresentOrElse(
+                gym1 -> gymMemberRepository1.save(gymMember1),
+                () -> result.set("Gym not found!")
+                );
+
+        return result;
+    }
+
+    public AtomicReference<String> checkIn(int memberId, String checkIn) throws ParseException {
+
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-mm-dd hh:mm:ss");
+        Date checkInGym = (Date) sdf.parse(checkIn);
+        AtomicReference<String> result = new AtomicReference<>("Member checked in!");
+        Timestamps1 timestamps1 = new Timestamps1(0, memberId, checkInGym, null);
+        Optional<GymMember1> gymMember = gymMemberRepository1.findById(memberId);
+        gymMember.ifPresentOrElse(
+                gymMember1 -> timestampsRepository1.save(timestamps1),
+                () -> result.set("Member not found!")
```

Comment:
I like the approach here, but ideally, if a member is not found the API should return a 404 (or similar type of status code).    Let's keep this for now and consider modifying the API to return a 404 when a member id is not found.